### ISSUE TITLE
path is too long

### ISSUE THREAD

**Issue by infofromca** [OrchardCMS/Orchard#5446]
when install it from dev or 1.9.x, 1.9.1 today's version,
got this error. (visual studio 2013)
![too-long-gulp](https://cloud.githubusercontent.com/assets/297675/8394588/35b66822-1d0d-11e5-9749-da5f63245132.png)
"~/Modules/Orchard.DynamicForms/node_modules/gulp-less/node_modules/gulp-util/node_modules/dateformat/node_modules/meow/node_modules/indent-string/node_modules/repeating/node_modules/is-finite/node_modules/"

**Comment by infofromca** [comment]
http://blog.icanmakethiswork.io/2014/12/gulp-npm-long-paths-and-visual-studio-fight.html

**Comment by sebastienros** [comment]
I know it has been fixed in the latest npm version.

**Comment by bleroy** [comment]
Link?

**Comment by bleroy** [comment]
Duplicate: #5561

**Comment by Skrypt** [comment]
https://windows.uservoice.com/forums/265757-windows-feature-suggestions/suggestions/6573539-remove-the-long-standing-max-path-limitation

Vote :)

**Comment by Skrypt** [comment]
We should then automate the removal of those node_modules before publishing.
Or we could also excluse those folder from publishing.

http://blogs.msdn.com/b/webdev/archive/2010/04/22/web-deployment-excluding-files-and-folders-via-the-web-application-s-project-file.aspx

**Comment by Skrypt** [comment]
Not working either.

http://stackoverflow.com/questions/25237823/actually-excluding-files-from-publish-without-verifying-it

**Comment by Skrypt** [comment]
Added this in Post built event command line of the 3 projects that holds node_modules folders.

```
if exist "$(ProjectDir)node_modules"  ( rimraf $(ProjectDir)node_modules )`
```

It deletes the folders but that's not really an adequate solution. Though it fixes the publish problem automatically. You also need to do :

```
npm install rimraf -g
```
before for installing rimraf globally.

**Comment by DanielStolt** [comment]
This will be fixed as part of #5450.

**Comment by dcinzona** [comment]
Installing globally isn't really a viable solution for everyone using the source.  @DanielStolt what's your plan for getting this in the solution?  I mentioned this in the dupe issue but I had success creating a generic project and adding it to the solution, adding the gulpfile and package.json manifest to it.  This allows me to download all of the npm modules into a separate directory in the root and avoid deployment issues.  Just need to remember to add it to all solutions (Azure Cloud Service and Orchard.Web).

As you had mentioned before, including a git-ignore and vs-ignore config in there would be beneficial as to not commit those node modules to source control.

**Comment by DanielStolt** [comment]
@dcinzona This sounds like what I have in mind? Going to work on it this weekend.

**Comment by dcinzona** [comment]
Awesome! Looking forward to it 
Thanks @DanielStolt

**Comment by DanielStolt** [comment]
Fixed in 6da6c5c.

**Comment by gamicklisafisher** [comment]
You should check out ‘Long Path Tool’ it will help you fix the issues.
I had faced similar issue and this tool helped me to resolve this issue.
